src/api/tokenList/TokenListApi.ts

```diff
@@ -108,6 +122,55 @@ export class TokenListApiImpl extends GenericSubscriptions<TokenDetails[]> imple
     currentUserList.push(token)
     localStorage.setItem(storageKey, JSON.stringify(currentUserList))
   }
+
+  private persistTokensForNetwork(networkId: number): void {
+    const storageKey = TokenListApiImpl.getLocalStorageKey(networkId)
+    localStorage.setItem(storageKey, JSON.stringify(this._tokensByNetwork[networkId]))
+  }
+
+  private async updateTokenIds(networkId: number): Promise<void> {
+    // Set as updated on start to prevent concurrent queries
+    this.updatedIdsForNetwork.add(networkId)
+
+    // Sequentially fetch updated ids
+    const updatedTokenList = []
+    let failedToUpdate = false
+    for (let i = 0; i < this._tokensByNetwork[networkId].length; i++) {
+      const token = this._tokensByNetwork[networkId][i]
+      const tokenAddress = token.address
+
+      try {
+        token.id = await this.exchangeApi.getTokenIdByAddress({ networkId, tokenAddress })
```

Comment:
Aggregate tokenIds and do `getTokenIdByAddress` calls in `Promise.all` otherwise it will add up quickly

Reply:
That's what I was doing at first. Then, with all the failures I was getting (due to wrong network) I thought throttling was necessary... you know the story.

I decided to do it one by one to maybe avoid that? Not sure makes any difference though.

Alright, will do the bundled way :)

Reply:
:point_right:  https://github.com/gnosis/dex-react/pull/824/commits/e6c39f6217e0f46f98b0749edc3d0a4bab12425c

Honestly though, old school for loop was more efficient, looping through list only once. Now I'm looping it three times:
1. Prepare list of promises
2. Resolve promises
3. Use results

Or I am just really bad at this.